scripts/create.sh

```diff
@@ -138,76 +148,88 @@ if [[ ! "$(kubectl get clusterrolebinding --field-selector metadata.name=cluster
+  sed -i'' -e "s/CONTROL_PLANE_AUTH_POLICY=NONE/CONTROL_PLANE_AUTH_POLICY=${ISTIO_AUTH_POLICY}/g" "$ISTIO_DIR/cluster.env"
 fi
 
 # Generate the DNS configuration necessary to have the GCE VM join the mesh.
-./install/tools/setupMeshEx.sh generateDnsmasq
+(
+  cd "$ISTIO_DIR"
+  "$ISTIO_DIR/install/tools/setupMeshEx.sh" generateDnsmasq
+)
 
 # Create the namespace to be used by the service on the VM.
-kubectl apply -f ../namespaces.yaml
+kubectl apply -f "$ROOT/scripts/namespaces.yaml"
 
 # Create the keys for mTLS
-./install/tools/setupMeshEx.sh machineCerts default vm all
+(
+  cd "$ISTIO_DIR"
+  "$ISTIO_DIR/install/tools/setupMeshEx.sh" machineCerts default vm all
+)
 
 # Re-export the GCP_OPTS to switch the project to the project where the VM
 # resides
 export GCP_OPTS="--zone ${ZONE} --project ${GCE_PROJECT}"
 # Setup the Istio service proxy and service on the GCE VM
-./install/tools/setupMeshEx.sh gceMachineSetup "${GCE_VM}"
+(
+  cd "$ISTIO_DIR"
+  "$ISTIO_DIR/install/tools/setupMeshEx.sh" gceMachineSetup "${GCE_VM}"
+)
 
 # Mesh expansion completed
 
 # Register the external service with the Istio mesh
-istioctl register -n vm mysqldb "$(gcloud compute instances describe "${GCE_VM}" \
+"$ISTIO_DIR/bin/istioctl" register -n vm mysqldb "$(gcloud compute instances describe "${GCE_VM}" \
   --format='value(networkInterfaces[].networkIP)' --project "${GCE_PROJECT}" --zone "${ZONE}")" 3306
 
 # Install the bookinfo services and deployments and set up the initial Istio
 # routing. For more information on routing see this Istio blog post:
 # https://istio.io/blog/2018/v1alpha3-routing/
-kubectl apply -f ./samples/bookinfo/platform/kube/bookinfo.yaml
-kubectl apply -f ./samples/bookinfo/networking/bookinfo-gateway.yaml
-kubectl apply -f ./samples/bookinfo/networking/destination-rule-all-mtls.yaml
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/platform/kube/bookinfo.yaml"
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/networking/bookinfo-gateway.yaml"
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/networking/destination-rule-all-mtls.yaml"
 
 # Change the routing to point to the most recent versions of the bookinfo
 # microservices
-kubectl apply -f ./samples/bookinfo/networking/virtual-service-reviews-v3.yaml
-kubectl apply -f ./samples/bookinfo/platform/kube/bookinfo-ratings-v2-mysql-vm.yaml
-
-popd || exit
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/networking/virtual-service-reviews-v3.yaml"
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/networking/virtual-service-ratings-mysql-vm.yaml"
+kubectl apply -n default \
+  -f "$ISTIO_DIR/samples/bookinfo/platform/kube/bookinfo-ratings-v2-mysql-vm.yaml"
 
 # Install and deploy the database used by the Istio service
-gcloud compute ssh "${GCE_VM}" --project="${GCE_PROJECT}" \
-  --zone "${ZONE}" \
-  -- "$(cat setup-gce-vm.sh)"
+# shellcheck disable=SC2086
```

Comment:
Well, I figured it would come up in this review and you took the bait! :-)

If anyone can help me figure out how to escape the quotes in the `--command` argument I'd be happy to take this out entirely. Any ideas? I can get it to either function or pass the linter, but not both.

Reply:
I think it's OK to disable it after we invested lots of time trying to fix it. Just leave a comment so people know what the rule is that we're disabling.

Reply:
Ok